fastestimator/network.py

```diff
@@ -674,6 +693,10 @@ def _optimizer_fn_to_optimizer(optimizer_fn: Union[Callable, None], model: Model
     if optimizer_fn:
         if framework == "tf":
             optimizer = optimizer_fn()
+            # initialize optimizer variables
+            _ = optimizer.iterations
```

Comment:
I thought we had to do this after finding out about whether we are running multi-GPU or not

Reply:
this is not about multi-gpu system. initialization of optimizer variables is universal.